### PR TITLE
[Mri violations]Update react table mri protocol violations

### PR DESCRIPTION
**Pull request by kongtiaowang**

### CODE REVIEW

**Comment by kongtiaowang** [comment]
add gruntfile.js

**Comment by alisterdev** [comment]
@kongtiaowang Why not use standard `columnFormatter` name?

File: modules/mri_violations/php/NDB_Menu_Filter_mri_protocol_violations.class.inc [hunk 1]
```diff
@@ -154,5 +161,22 @@ class NDB_Menu_Mri_Protocol_Violations extends NDB_Menu_Filter
...
+     */
+    function getJSDependencies()
+    {
+        $factory = NDB_Factory::singleton();
+        $baseURL = $factory->settings()->getBaseURL();
+        $deps    = parent::getJSDependencies();
+        return array_merge(
+            $deps,
+            array(
+                $baseurl . "/mri_violations/js/mri_protocol_violations_columnFormatter.js",
```
Comment (alisterdev) [inline modules/mri_violations/php/NDB_Menu_Filter_mri_protocol_violations.class.inc]: `$baseURL ` is not spelled correctly

File: modules/mri_violations/templates/menu_mri_protocol_violations.tpl [hunk 2]
```diff
@@ -124,44 +124,19 @@
...
-    RowsPerPage : {$rowsPerPage},
-    Total: {$TotalItems},
-    onChangePage: function(pageNum) {
-        location.href="{$baseurl}/mri_violations/?submenu=mri_protocol_violations&filter[order][field]={$filterfield}&filter[order][fieldOrder]={$filterfieldOrder}&pageID=" + pageNum
-    },
-    Active: {$pageID}
-});
-React.render(pageLinks, document.getElementById("pageLinks"));
+loris.hiddenHeaders = {(empty($hiddenHeaders))? [] : $hiddenHeaders };
+var hasWritePermission = {json_encode($hasWritePermission)};
```
Comment (alisterdev) [inline modules/mri_violations/templates/menu_mri_protocol_violations.tpl]: `hasWritePermission` shouldnt be here

**Comment by kongtiaowang** [comment]
@alisterdev 
Could you review this?

**Comment by alisterdev** [comment]
@kongtiaowang As discussed with Dave, all mri_violations react updates will go into 17.1. So I'll test it after 17.0 release

**Comment by alisterdev** [comment]
Also this needs rebase

**Comment by alisterdev** [comment]
@kongtiaowang Can you pull latest changes from 17.1, rebase this PR and fix stuff from review. 
We could retest and merge it then.

**Comment by kongtiaowang** [comment]
@alisterdev  
I rebased this branch into #2523.
Could u review the new PR?

**Pull request closed without merging**